libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -214,9 +216,28 @@ instance BaseUniverse fn => Functions (MapFn fn) fn where
                       Set.empty
                       must
                       size
-                      (constrained $ \kv -> satisfies (app (sndFn @fn) (app (toGenericFn @fn) kv)) elemspec)
+                      -- (constrained $ \kv -> satisfies (app (sndFn @fn) (app (toGenericFn @fn) kv)) elemspec)
+                      (constrained $ \kv -> satisfies (snd_ kv) elemspec)
                       foldspec
                 _ -> ErrorSpec ["Rng on bad map spec", show spec]
+    RngSet ->
+      case fn of
+        (_ :: MapFn fn '[Map k v] (Set v))
+          | NilCtx HOLE <- ctx
+          , Evidence <- prerequisites @fn @(Map k v) ->
+              case spec of
+                MemberSpec [r] ->
+                  typeSpec $ MapSpec Nothing Set.empty (Set.toList r) (equalSpec $ sizeOf r) TrueSpec NoFold
+                TypeSpec (SetSpec must elemspec _size) [] ->
```

Comment:
Ignoring the `size` here is not sound.

Reply:
The problem with making this sound is that you have to deal with the case where there are positive lower bounds on the size. This in turn means that the size constrains the number of _different_ elements in the range.

Reply:
Might we try this:
```
TypeSpec (SetSpec must elemspec setsize) [] ->
                  typeSpec $
                    MapSpec
                      Nothing
                      Set.empty
                      (Set.toList must)
                      ((geqSpec (fromIntegral (Set.size must))) <> setsize) -- This might be overconstrained.
                      (constrained $ \kv -> satisfies (app (sndFn @fn) (app (toGenericFn @fn) kv)) elemspec)
                      NoFold
```
I would reason as follows. The set must contain the elements in `must` so the smallest the mat can be is the length of `must`.  It may contain other things, So if it does then`mapsize` constrains how many. So if the SetSpec is consistent, the new MapSpec should be too.  Is there something else I am missing here?

Reply:
I think it would be a good idea to use `prop_sound` to test your ideas here. A good basic test is:
```haskell
rngSpec :: Specification BaseFn (Map Int Int)
rngSpec = constrained $ \ m -> 3 <=. sizeOf_ (rngSet_ m)
```

Reply:
Ok I added this test, made every thing work. So this should be resolved